File: momen-android/core/src/main/kotlin/com/ahmadtambaya/momen/core/ClipNaming.kt
package com.ahmadtambaya.momen.core

/**
 * Clip naming — a project sets a prefix once, and clips auto-number from it:
 * "MONTACLIP" → MONTACLIP_001, MONTACLIP_002, … (3-digit zero-padded).
 * Kept in lockstep with momen-ios/MomenKit/.../ClipNaming.swift.
 */
object ClipNaming {

    const val SEPARATOR = "_"

    /** Keep letters/digits/_/-, drop everything else, strip a trailing sep. */
    fun normalizePrefix(raw: String): String {
        var prefix = raw.filter { it.isLetterOrDigit() || it == '_' || it == '-' }
        while (prefix.endsWith(SEPARATOR)) prefix = prefix.dropLast(1)
        return prefix.ifEmpty { "CLIP" }
    }

    /** Build a clip name from a normalized prefix and a 1-based number. */
    fun name(prefix: String, number: Int): String =
        "%s%s%03d".format(java.util.Locale.ROOT, prefix, SEPARATOR, number)
}

File: momen-ios/MomenKit/Sources/MomenKit/ClipNaming.swift
import Foundation

/// Clip naming — a project sets a prefix once, and clips auto-number from it:
/// "MONTACLIP" → MONTACLIP_001, MONTACLIP_002, … (3-digit zero-padded).
public enum ClipNaming {

    public static let separator = "_"

    /// Normalise a user-entered prefix: keep letters/digits/_/-, drop
    /// everything else (spaces included), strip a trailing separator.
    /// Empty result falls back to "CLIP".
    public static func normalizePrefix(_ raw: String) -> String {
        var prefix = String(raw.filter { ch in
            ch.isLetter || ch.isNumber || ch == "_" || ch == "-"
        })
        while prefix.hasSuffix(separator) { prefix.removeLast() }
        return prefix.isEmpty ? "CLIP" : prefix
    }

    /// Build a clip name from a normalized prefix and a 1-based number.
    public static func name(prefix: String, number: Int) -> String {
        "\(prefix)\(separator)\(String(format: "%03d", number))"
    }
}
